### PR TITLE
Introducing search count feature

### PR DESCRIPTION
**Pull request by anandraj346**
Previously on writing into the search input text, we were not shown how many matches found for that string. Introducing the search count feature will help user address how many counts of the search string found.

https://github.com/user-attachments/assets/bae6bcb9-3bca-4775-885a-efdd7d3855ab

### CODE REVIEW

**Review by badasswp — CHANGES_REQUESTED** [review]
@anandraj346 I can see duplicity of effort & code in trying to implement this search count feature in this PR.

Since we already have those functions implemented, what we should be doing is passing in a `context` param that represents either a `true` (**replace**) state or `false` (**search**) state and then use that conditional in our `replaceBlockAttribute` callback.

Let's pass in the `context` param here:
```js
  const replace = (context = false): void => {
    setContext(context);
    setReplacements(0);
```

On this line, pass `context` in:
```js
    select('core/block-editor').getBlocks().forEach((element) => {
      recursivelyReplace(element, pattern, replaceInput, context);
    });
```

Do the same for other functions until we get to the `replaceBlockAttribute` callback. In this method, let's check for the context value and if it is TRUE, then let's replace.
```js
    if (context) {
      (dispatch('core/block-editor') as any).updateBlockAttributes(clientId, property);
    }
```

Same here:
```js
    // Handle edge-case ('value') with Pullquotes.
    if (attributes.value) {
      if (context) {
        (dispatch('core/block-editor') as any)
          .updateBlockAttributes(clientId, { value: newString });
      }
      setReplacements((items) => items + 1);
    }
```

**Comment by badasswp** [comment]
@anandraj346 Could you pls create an issue for this as well. Thanks.

File: src/app.tsx [hunk 1]
```diff
@@ -249,6 +434,16 @@ const SearchReplaceForBlockEditor = () => {
...
+                  <p>
+                    <strong>{matchCount}</strong> {__('item(s) found', 'search-replace-for-block-editor')}.
+                  </p>
+                </div>
+              ) : ''
+            }
+
             <div id="search-replace-modal__button-group">
               <Button
                 variant="primary"
```
Comment (badasswp) [inline src/app.tsx:449]: @anandraj346 Let's refactor the `onClick` logic to this:

```js
                onClick={() => replace(true)}
```

File: src/core/app.tsx [hunk 2]
```diff
@@ -260,7 +281,7 @@ const SearchReplaceForBlockEditor = (): JSX.Element => {
             </div>
 
             {
-              replacements ? (
+              context ? (
```
Comment (badasswp) [inline src/core/app.tsx:284]: @anandraj346 I see duplication here, let's refactor this to:

```js
            {
              replacements ? (
                <div id="search-replace-modal__notification">
                  <p>
                    {context ? (
                      <>
                        <strong>{replacements}</strong> {__('item(s) replaced successfully', 'search-replace-for-block-editor')}.
                      </>
                    ) : (
                      <>
                        <strong>{replacements}</strong> {__('item(s) found', 'search-replace-for-block-editor')}.
                      </>
                    )}
                  </p>
                </div>
              ) : ''
            }
```

File: src/core/app.tsx [hunk 3]
```diff
@@ -269,10 +290,20 @@ const SearchReplaceForBlockEditor = (): JSX.Element => {
               ) : ''
             }
 
+            {
```
Comment (badasswp) [inline src/core/app.tsx:293]: @anandraj346 Pls remove this.

Comment (anandraj346) [reply src/core/app.tsx]: Resolved

**Review by badasswp — APPROVED** [review]
@anandraj346 This looks good 👍. Thanks for all your work on this.

feature-search-count

**Pull request merged**